lib/mayaUsd/render/vp2RenderDelegate/draw_item.h

```diff
@@ -38,19 +39,18 @@ class HdVP2RenderDelegate;
 class HdVP2DrawItem final : public HdDrawItem
 {
 public:
-    //! A primvar vertex buffer map indexed by primvar name.
-    using PrimvarBufferMap = std::
-        unordered_map<TfToken, std::unique_ptr<MHWRender::MVertexBuffer>, TfToken::HashFunctor>;
-
     //! Helper struct providing storage for render item data
     struct RenderItemData
     {
-        //! Render item color buffer - use when updating data
-        std::unique_ptr<MHWRender::MVertexBuffer> _colorBuffer;
-        //! Render item normals buffer - use when updating data
-        std::unique_ptr<MHWRender::MVertexBuffer> _normalsBuffer;
-        //! Render item primvar buffers - use when updating data
-        PrimvarBufferMap _primvarBuffers;
+        //! Unique name for easier debugging and profiling.
+        MString _renderItemName;
+        //! Pointer of the render item for fast access. No ownership is held.
+        MHWRender::MRenderItem* _renderItem { nullptr };
+
+        //! The geom subset this render item represents. _geomSubset.id is StdPath::EmptyPath() if
+        //! there is no geom subset.
+        HdGeomSubset _geomSubset;
```

Comment:
Do we have to maintain another instance here? If the geom subset is stored in the topology, is it safe for us to just store a pointer here?

Reply:
It is not 100% clear if a pointer is safe. The real question is when the topology gets dirtied how much of it changes? Are all the HdGeomSubset objects destroyed and re-created? When topology changes will we try to re-use existing HdGeomSubset items or destroy them all and start again from scratch?

I don't think we have workflows that modify topology, so all I have a questions and not answers.

The cost of copying the HdGeomSubset is low because it holds only SdfPath and VtIntArray objects, both of which have fast copy operations.